crates/turborepo-lib/src/package_changes_watcher.rs

```diff
@@ -167,121 +230,133 @@ impl Subscriber {
+                    continue;
+                }
+
+                let ChangedFiles::Some(trie) = changed_files.take() else {
+                    let _ = self
+                        .package_change_events_tx
+                        .send(PackageChangeEvent::Rediscover);
 
-                                for pkg in changed_pkgs {
-                                    let _ = self.package_change_events_tx.send(
-                                        PackageChangeEvent::Package {
-                                            name: pkg.name.clone(),
-                                        },
-                                    );
+                    match self.initialize_repo_state().await {
+                        Some((new_repo_state, new_gitignore)) => {
+                            repo_state = new_repo_state;
+                            root_gitignore = new_gitignore;
+                            change_mapper = match repo_state.get_change_mapper() {
+                                Some(change_mapper) => change_mapper,
+                                None => {
+                                    break;
                                 }
-                            }
-                            Err(err) => {
-                                // Log the error, rediscover the packages and try again
-                                tracing::error!("error: {:?}", err);
-
-                                let _ = self
-                                    .package_change_events_tx
-                                    .send(PackageChangeEvent::Rediscover);
-                                match self.initialize_repo_state().await {
-                                    Some(new_repo_state) => {
-                                        repo_state = new_repo_state;
-                                        change_mapper = match repo_state.get_change_mapper() {
-                                            Some(change_mapper) => change_mapper,
-                                            None => {
-                                                break;
-                                            }
-                                        }
-                                    }
+                            };
+                        }
+                        None => {
+                            break;
+                        }
+                    }
+                    continue;
+                };
+
+                let gitignore_path = self.repo_root.join_component(".gitignore");
+                if trie.get(gitignore_path.as_std_path()).is_some() {
+                    let (new_root_gitignore, _) = Gitignore::new(&gitignore_path);
+                    root_gitignore = new_root_gitignore;
+                }
+
+                let changed_files: HashSet<_> = trie
+                    .keys()
+                    .filter_map(|p| {
+                        let p = AbsoluteSystemPathBuf::try_from(p.as_path()).ok()?;
```

Comment:
Should this be `.expect("filewatching returns absolute paths")`?

Reply:
Do we want to crash the daemon on this? That seems a little drastic